### ISSUE TITLE
Whitelist filtering

### ISSUE THREAD

**Issue by bamueh** [acorg/dark-matter#784]
I was under the impression that passing a `whitelist` to `filter-fasta.py` would remove all sequences except those that are in the `whitelist`. Instead, it seems like nothing gets filtered at all. But maybe I'm misunderstanding something?

**Comment by terrycojones** [comment]
How did you pass a whitelist? There are two options, `--whitelist` and `--whitelistFile`.  If you look in `test/test_filter.py` you'll see quite a few whitelist tests.  I'm guessing you did `--whitelist FILE` instead of `--whitelistFile FILE` ?   (We could easily make a change to detect this and give a warning.)

**Comment by bamueh** [comment]
No, I did `--whitelist somesequencename` and `--whitelistFile FILE` and both didn't work.

**Comment by bamueh** [comment]
e.g. 
`$ filter-fasta.py --whitelist OP866294_C_C2.2_Ethiopia_Camelus_dromedarius_2019-11-23 < data/sequences.fasta > sss.fasta
Read 640 sequences, kept 640 (100.00%).`

where `OP866294_C_C2.2_Ethiopia_Camelus_dromedarius_2019-11-23` is a name of a sequence in `data/sequences.fasta`.

**Comment by terrycojones** [comment]
OK, got it. A whitelist is used to guarantee that some things don't get thrown away. A blacklist ensures that some things do. The whitelist has priority. So what you should do is:

```sh
$ filter-fasta.py --whitelist OP866294_C_C2.2_Ethiopia_Camelus_dromedarius_2019-11-23 \
    --negativeTitleRegex '.*' < data/sequences.fasta > sss.fasta
```

I.e., you ask that everything be thrown away, _except_ the ones on the whitelist.

**Comment by terrycojones** [comment]
BTW, these terms are now considered racist.

**Comment by terrycojones** [comment]
If you look at the tests, you'll see several that have `negativeRegex="."`
